### PR TITLE
whitelisting script

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: scripts/uploadWhitelist.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+};
+const isCurrency = (currency, address) => {
+  if (currency !== "EUR" && currency !== "ETH")
+    throw new Error(
+      `Investor:${address} and currency:${currency} has wrong currency format!!`
+    );
+  return tokenEnum[currency];
+};
+const getAmount = (amount, address) => {
+  const investAmount = Number.parseFloat(amount.replace(/^\D+/g, ""));
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: we need to use better parser. this one accepts any messy string starting with a number

File: scripts/uploadWhitelist.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+const getAttributes = (address, currency, amount) => ({
+  wlAddresses: isAddress(address),
+  wlTokens: isCurrency(currency, address),
+  wlTickets: getAmount(amount, address)
+});
+const isDuplicate = array => {
+  const duplicateArray = array.filter(
+    data =>
+      array.filter(investor => {
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: use `some`, `undefined` should not be returned

File: scripts/uploadWhitelist.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  }
+};
+const filterWlfromSmartContract = async (filteredWhiteList, commitment) => {
+  let lastUploadedfile = false;
+  const verifiedWhiteList = (await Promise.all(
+    filteredWhiteList.map(async investor => {
+      const whiteListTicketbyAddress = await commitment.whitelistTicket(
+        investor.wlAddresses
+      );
+      const tokenType = whiteListTicketbyAddress[0].isZero();
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: what is `isZero()`?

File: scripts/uploadWhitelist.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      const ticketSize = whiteListTicketbyAddress[1].toString();
+      const ticketSizefromList =
+        investor.wlTokens === 1
+          ? (await commitment.convertToEur(investor.wlTickets)).toString()
+          : investor.wlTickets.toString();
+
+      if (ticketSize !== ticketSizefromList && tokenType === false) {
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: use bigNumber.eq() not !==

Comment (rudolfix) [inline scripts/uploadWhitelist.js]: `tokenType === false` what does it mean here. if tokenType === 0 we should immediately show that expected investor is not on the list

File: scripts/uploadWhitelist.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    })
+  )).filter(investor => {
+    if (investor !== undefined) {
+      lastUploadedfile = true;
+    }
+    if (investor === undefined) {
+      if (lastUploadedfile === true)
+        throw new Error("Something went really wrong with the smart contracts");
+    }
+    return !!investor;
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: what !!means in javascript? two negations?

File: scripts/uploadWhitelist.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+      commitment
+    );
+    do {
+      const whitelistPayloud = verifiedWhiteList.slice(0, payloadSize);
+      verifiedWhiteList = verifiedWhiteList.slice(
+        payloadSize,
+        verifiedWhiteList.length
+      );
+
+      console.log("Setting whitelist Payloud");
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: Payload

File: scripts/uploadWhitelist.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+      const formattedPayload = formatPayload(whitelistPayloud);
+      console.log("Sending Payload to SmartContract");
+      const transactionReceipt = (await commitment.addWhitelisted(
+        formattedPayload[0].wlAddresses,
+        formattedPayload[1].wlTokens,
+        formattedPayload[2].wlTickets
+      )).receipt;
+      if ((await transactionReceipt.status) === 0)
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: always show receipt

File: scripts/uploadWhitelist.js [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+            ticketSizefromList
+          } token ${tokenType} for ${investor.wlAddresses}`
+        );
+      }
+      return tokenType ? investor : undefined;
+    })
+  )).filter(investor => {
+    if (investor !== undefined) {
+      lastUploadedfile = true;
+    }
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: why not else here?

File: scripts/uploadWhitelist.js [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+    if (investor === undefined) {
```
Comment (rudolfix) [inline scripts/uploadWhitelist.js]: what this one means? if we added an investor and now we see that someone is already added, we throw an error?

Comment (Moejoe90) [reply scripts/uploadWhitelist.js]: Of course this makes sure that the list is in order. 
If we find someone already added before there que an error is thrown

File: scripts/uploadWhitelist.js (hunk 5 above)
Comment (Moejoe90) [reply scripts/uploadWhitelist.js]: Its automatically done by the linter if not negetive return true

**Pull request merged**